### PR TITLE
test: Add test that worker status is what we expect

### PR DESCRIPTION
**Pull request by YutongLi291**
### What was the problem/requirement? (What/Why)
The worker agent has statuses that are exposed to users of Deadline Cloud. We should make sure that  the worker status returned by the API corresponds to the status that we expect the worker to be in, when we update the worker to the corresponding status.

We should make sure that code quality for this logic is kept high, so as to ensure a good customer experience in regards to the worker.
### What was the solution? (How)
Add testing regarding the worker.

Also moved some files so that another test for a test job with job Attachments is ran on both windows and Linux

### What is the impact of this change?
Better code quality
### How was this change tested?
`hatch run cross-os-e2e-tests`, `hatch run linux-e2e-tests`
### Was this change documented?
no
### Is this a breaking change?
no
----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: test/e2e/cross_os/test_job_submissions.py [hunk 1]
```diff
@@ -430,3 +439,83 @@ def test_worker_streams_logs_to_cloudwatch(
         )
 
         assert len(worker_logs["events"]) > 0
+
+    def test_worker_uses_job_attachment_configuration(
```
Comment (YutongLi291) [inline test/e2e/cross_os/test_job_submissions.py:443]: This is moved from the linux test file, so that it will run on both linux and windows

**Review by AlexTranAmz — APPROVED** [review]

File: test/e2e/cross_os/test_job_submissions.py [hunk 2]
```diff
@@ -430,3 +439,83 @@ def test_worker_streams_logs_to_cloudwatch(
...
+        job_id: Optional[str] = api.create_job_from_job_bundle(
+            job_bundle_path,
+            job_parameters,
+            priority=99,
+            config=config,
+            queue_parameter_definitions=[],
+        )
+        assert job_id is not None
+
+        job_details = Job.get_job_details(
```
Comment (jairaws) [inline test/e2e/cross_os/test_job_submissions.py:474]: nit: type

File: test/e2e/cross_os/test_job_submissions.py [hunk 3]
```diff
@@ -430,3 +439,83 @@ def test_worker_streams_logs_to_cloudwatch(
...
+        )
+        job = Job(
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            template={},
+            **job_details,
+        )
+        job.wait_until_complete(client=deadline_client, max_retries=20)
+
+        job_attachment_settings = get_queue(
```
Comment (jairaws) [inline test/e2e/cross_os/test_job_submissions.py:488]: nit: type

File: test/e2e/cross_os/test_job_submissions.py [hunk 4]
```diff
@@ -430,3 +439,83 @@ def test_worker_streams_logs_to_cloudwatch(
...
+            farm_id=deadline_resources.farm.id,
+            queue_id=deadline_resources.queue_a.id,
+        ).jobAttachmentSettings
+
+        assert job_attachment_settings is not None
+
+        job_output_downloader = download.OutputDownloader(
```
Comment (jairaws) [inline test/e2e/cross_os/test_job_submissions.py:495]: nit: type

File: test/e2e/cross_os/test_job_submissions.py [hunk 5]
```diff
@@ -430,3 +439,83 @@ def test_worker_streams_logs_to_cloudwatch(
...
+            s3_settings=job_attachment_settings,
+            farm_id=deadline_resources.farm.id,
+            queue_id=deadline_resources.queue_a.id,
+            job_id=job.id,
+            step_id=None,
+            task_id=None,
+        )
+
+        output_paths_by_root = job_output_downloader.get_output_paths_by_root()
```
Comment (jairaws) [inline test/e2e/cross_os/test_job_submissions.py:504]: nit: type

File: test/e2e/cross_os/test_worker_status.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+from deadline_worker_agent.api_models import (
+    WorkerStatus,
+)
+import backoff
+from deadline_test_fixtures import DeadlineClient, EC2InstanceWorker
+from utils import get_operating_system_name
+import pytest
+
+
+@pytest.mark.parametrize("operating_system", [get_operating_system_name()], indirect=True)
```
Comment (jairaws) [inline test/e2e/cross_os/test_worker_status.py:17]: are we planning on adding more parameters later on? Also, I'm not familiar with using parameters at test class level. What's the purpose of this? Could it be a fixture instead?

**Review by jairaws — APPROVED** [review]

**Pull request merged**